src/Arcus.Messaging.ServiceBus.Core/Extensions/MessageExtensions.cs

```diff
@@ -1,23 +1,33 @@
-﻿using Arcus.Messaging.Abstractions;
-using GuardNet;
+﻿using System;
+using System.Collections.Generic;
 
 // ReSharper disable once CheckNamespace
 namespace Microsoft.Azure.ServiceBus
 {
     public static class MessageExtensions
     {
         /// <summary>
-        ///     Gets the transaction id that is linked to this message
+        ///     Gets the user property with a given <paramref name="key"/>
         /// </summary>
-        /// <param name="message">Message to process</param>
-        /// <returns>Transaction id for message</returns>
-        public static string GetTransactionId(this Message message)
+        /// <typeparam name="T"></typeparam>
+        /// <param name="message"></param>
+        /// <param name="key"></param>
+        /// <returns></returns>
+        public static T GetUserProperty<T>(this Message message, string key)
```

Comment:
How about using `TProperty`? `T` is always too vague
```suggestion
        public static TProperty GetUserProperty<TProperty>(this Message message, string key)
```

Reply:
Ok!